target/linux/ath79/generic/base-files/lib/upgrade/openmesh.sh

```diff
@@ -0,0 +1,114 @@
+# The U-Boot loader of the OpenMesh devices requires image sizes and
+# checksums to be provided in the U-Boot environment.
+# The OpenMesh devices come with 2 main partitions - while one is active
+# sysupgrade will flash the other. The boot order is changed to boot the
+# newly flashed partition. If the new partition can't be booted due to
+# upgrade failures the previously used partition is loaded.
+
+trim()
+{
+	echo $1
+}
+
+cfg_value_get()
+{
+	local cfg=$1 cfg_opt
+	local section=$2 our_section=0
+	local param=$3 our_param=
+
+	for cfg_opt in $cfg
+		do
+			[ "$cfg_opt" = "[$section]" ] && our_section=1 && continue
+			[ "$our_section" = "1" ] || continue
+
+			our_param=$(echo ${cfg_opt%%=*})
```

Comment:
Is there a reason for this echo?

Reply:
You would have to ask Gabor Juhos about this

Reply:
The only reason I can think of would be to remove newlines (and convert them to spaces).

But I can live with just keeping this for now.

Reply:
This is rather unlikely because each entry of cfg_opt should not contain any newlines. I will have a look and may drop it

Reply:
Older OpenWrt is really crowded with useless echos. You actually removed several in the script already. So, probably, there really is no purpose after all.

But if you are unsure, as stated, I can live with keeping this one for now.

Reply:
Consider it dropped